### PR TITLE
Blog app integration specs

### PR DESCRIPTION
**Pull request by ometman**
We have completed the 'integration specs' of the project on the branch 'Blog_app_integration_specs':
- It was done through pair programming
- The N+1 problem was solved when fetching all posts and their comments for a user.
--- Before the fix:
beforen+1
---After the fix:
after n1

- An Integration Specs was written, using the 'Capybara' gem' for the following:
- -- User Index page
- --User show page
- --Use post index page
- --Post show page
-  Notice for this project requirement i for time i fixed just the testing file for integration you can check it by **rspec spec/features** and i will fix the other tests in next time.

### CODE REVIEW

**Review by OyePriscilla — COMMENTED** [review]
Hi @ometman 👍 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

Please @ometman kindly take note that you are supposed to use your repo for review, not your partner's repo. Ensure that your partner contribute meaningfully to your PR as well. Thank you. The guideline is screenshot below 👇 

Design1

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/features/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    it "should redirect me to that post's show page when clicking on the post's title" do
+      visit user_posts_path(@user1)
+      click_link @post1.title
+      expect(page).to have_current_path(user_post_path(@user1, @post1))
+    end
+
+    it 'Should display the section to create a new post' do
+      visit user_posts_path(@user1)
+      expect(page).to have_content('Create New Post')
+    end
```
Comment (Ibizugbe) [inline spec/features/post_index_spec.rb:67]: Nice job so far😉. In fulfillment of the[ requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md) for this milestone, kindly add a test that asserts that you can see a section for pagination if there are more posts than fit on the view😉.

image

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @newtonah0 👋🏼👋🏼👋🏼👋🏼, 

require_changes


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] Linters are passing👍🏼.
- [x] gitflow is used👍🏼.
- [x] Test cases are properly written for the user index and show page👍🏼.


## Required Changes ♻️

- [x] Kindly note that a couple of your tests are failing and as a requirement, all tests should be passing. I suggest you ensure all your tests are passing before you proceed to submit, to avoid any form of blocker👍🏼.
image


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mckent05 — APPROVED** [review]
Hi @ometman , 

# STATUS: APPROVED ✔️ ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations and well done! 🎉 
approved


## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👏 👏 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**